src/model.c

```diff
@@ -255,6 +287,115 @@ static bool parse_int(char **buffer, int *i)
     return true;
 }
 
+struct model *model_load_from_ply(const char *fname)
+{
+    FILE *fp = fopen(fname, "r");
+    if (!fp)
+    {
+        fprintf(stderr, "ERROR: failed to load file \"%s\".\n", fname);
+        return NULL;
+    }
+
+    // Create a new model
+    struct model *model = model_init();
+
+    // Read each line of the file
+    char buffer[256];
+
+    int vertex_count = 0;
+    int face_count = 0;
+
+    while (fgets(buffer, sizeof(buffer), fp))
+    {
+        char *p = buffer;
+        while (*p)
+        {
+            if (*p == '\n' || *p == '\r')
+                *p = '\0';
+            if (*p == '\t')
+                *p = ' ';
+            p++;
+        }
+
+        char *bufferp = buffer;
+        char *instr = str_chop_skip_empty(&bufferp, " ");
+
+        if (!instr || instr[0] == '#')
+            continue;
+
+        if (strcmp(instr, "element") == 0)
+        {
+            char *type = str_chop_skip_empty(&bufferp, " ");
+            int count = 0;
+            parse_int(&bufferp, &count);
+
+            if (strcmp(type, "vertex") == 0)
+            {
+                model_reserve_vertexes(model, count);
+                vertex_count = count;
+            }
+            else if (strcmp(type, "face") == 0)
+            {
+                model_reserve_faces(model, count);
+                face_count = count;
+            }
+        }
+        else if (strcmp(instr, "property") == 0)
+        {
+            // Skip property declarations
+        }
+        else if (strcmp(instr, "end_header") == 0)
+        {
+            // End of header, start reading data
+            break;
+        }
+    }
+
+    for (int i = 0; i < vertex_count; i++)
+    {
+        float f1, f2, f3;
+
+        if (!fgets(buffer, sizeof(buffer), fp) ||
+            sscanf(buffer, "%f %f %f", &f1, &f2, &f3) != 3)
+        {
+            fprintf(stderr, "ERROR: invalid vertex data.\n");
+            fclose(fp);
+            model_free(model);
+            return NULL;
+        }
+
+        vec3 vec;
+        vec.x = f1;
+        vec.y = f2;
+        vec.z = f3;
```

Comment:
I noticed that the models appear rotated so probably this is not the correct ordering for the axis. Perhaps vec.y should be f3 and vec.z should be f2 ?

Reply:
Actually, the ordering of the axes came inconsistent between the models PLY models. So the teapot.ply and the airplane.py had the wrong orientation.

Perhaps we should stick to the axes ordering used by blender when exporting .PLY files.